File: internal/models/transaction.go
package models

type Type string
type PaymentMethod string

const (
	TypeIncome  Type = "INCOME"
	TypeExpense Type = "EXPENSE"

	PaymentMethodCreditCard PaymentMethod = "CREDIT_CARD"
	PaymentMethodDebitCard  PaymentMethod = "DEBIT_CARD"
	PaymentMethodPix        PaymentMethod = "PIX"
)

type Transaction struct {
	TransactionId         string        `json:"transactionId" dynamo:"TransactionId"`
	CategorySubcategoryId string        `json:"-" dynamo:"CategorySubcategoryId"`
	Category              string        `json:"category" dynamo:"Category"`
	Value                 float64       `json:"value" dynamo:"Value"`
	Description           string        `json:"description" dynamo:"Description"`
	MonthYear             string        `json:"monthYear" dynamo:"MonthYear"`
	Date                  string        `json:"date" dynamo:"Date"`
	Type                  Type          `json:"type" dynamo:"Type"`
	PaymentMethod         PaymentMethod `json:"paymentMethod" dynamo:"PaymentMethod"`
}

File: web/fm-finances/src/models/transaction.ts
import { CategoryModel } from "./category";

export interface TransactionModel {
    transactionId: string;
    description: string;
    monthYear: string;
    value: number;
    paymentMethod: string,
    type: string;
    category: string;
    date: string;
}

export interface TransactionPageModel {
    results: TransactionModel[],
    next: string,
}